### DISCUSSION TITLE
Is it possible to change result data and don't change allure source code?

### DISCUSSION THREAD

**Discussion by stcnchenxin** [allure-framework/allure2#1702] [General Discussion]
I am trying to test an app, and when something goes wrong, it will snapshot, and record the click position. When generating the report, I hope that the report can show the click position (like a circle attached to the pitcher).

Now I do this by modified the source code of `allure-generator` and `allure-plugin-api` (to rocode the position in Attachment), and rebuild jar file, replace same file in allure commandline tools.

Can i do this without change the source code? 

Is that writing a plugin  a solution? And how to do this?

**Comment by cheshi-mantu** [comment]
@stcnchenxin hi. The report is only showing it receives from your test results.
It's a task for your code to provide the data with the click position to the report tool.
here is an example of such a solution: https://selenide.org/2019/12/06/advent-calendar-visualize-click/

**Reply by stcnchenxin** [reply]
Thanks, it is helpful for me.
